rq_win/worker.py

```diff
@@ -80,13 +83,24 @@ def perform_job(self, job, queue):
             job._status = rq.job.JobStatus.FINISHED
             job.ended_at = times.now()
 
-            result_ttl = job.get_ttl(self.default_result_ttl)
-            pipeline = self.connection._pipeline()
-            if result_ttl != 0:
-                job.save(pipeline=pipeline)
-            queue.enqueue_dependents(job, pipeline=pipeline)
-            job.cleanup(result_ttl, pipeline=pipeline)
-            pipeline.execute()
+            #
+            # Using the code from Worker.handle_job_success
+            #
+            with self.connection.pipeline() as pipeline:
+                pipeline.watch(job.dependents_key)
+                queue.enqueue_dependents(job, pipeline=pipeline)
+
+                self.set_current_job_id(None, pipeline=pipeline)
+                self.increment_successful_job_count(pipeline=pipeline)
+
+                result_ttl = job.get_result_ttl(self.default_result_ttl)
+                if result_ttl != 0:
+                    job.save(pipeline=pipeline, include_meta=False)
+
+                job.cleanup(result_ttl, pipeline=pipeline,
+                            remove_from_queue=False)
+
+                pipeline.execute()
```

Comment:
This way of extending Worker looks outdated in more ways than the parameters for `work` and the success handling. Are there more new behaviors from the base class that need to be accounted for?

Reply:
Honestly, I don't know -- I actually started by patching `rq` directly to work on Windows (which I succeeded in) and then I found your code, which was much more elegant but didn't work with `rq==1.0`.

I wanted to make the minimal amount of changes to get this off the ground. This PR was sufficient to get `rq` working on Windows (which solves a problem for me).

I tested this, and it works for me (for both simple and more involved examples). However, I haven't studied things in-depth: this really was the MVP.

Reply:
Ok